pkg/analytics/opt_test.go

```diff
@@ -19,20 +20,25 @@ func TestSetOptStr(t *testing.T) {
 	defer f.tearDown()
 	f.assertOptStatus(analytics.OptDefault)
 
-	analytics.SetOptStr("opt-in")
-	f.assertOptStatus(analytics.OptIn)
-
-	analytics.SetOptStr("opt-out")
-	f.assertOptStatus(analytics.OptOut)
-
-	analytics.SetOptStr("in")
-	f.assertOptStatus(analytics.OptIn)
-
-	analytics.SetOptStr("out")
-	f.assertOptStatus(analytics.OptOut)
+	for _, v := range []struct {
+		s string
+		opt analytics.Opt
+	}{
+		{"opt-in", analytics.OptIn},
+		{"opt-out", analytics.OptOut},
+		{"in", analytics.OptIn},
+		{"out", analytics.OptOut},
+	} {
+		opt, err := analytics.SetOptStr(v.s)
+		if !assert.NoError(t, err) {
+			t.FailNow()
+		}
+		assert.Equal(t, v.opt, opt)
+		f.assertOptStatus(v.opt)
+	}
 
-	analytics.SetOptStr("foo")
-	f.assertOptStatus(analytics.OptDefault)
+	_, err := analytics.SetOptStr("foo")
+	assert.Errorf(t, err, "asdf")
```

Comment:
that's a very descriptive message you've got here for this assert...

Reply:
oh, oops. generally when asserting an error string I'll just put in asdf to see what it's looking for when it fails, but apparently that string isn't actually tested? ugh